tests/test.VmAgentRealVmadm.js

```diff
@@ -841,6 +842,136 @@ test('Real vmadm, fake VMAPI: validate DNI', function _test(t) {
+            coordinator.once('vmapi.updateServerVms',
+                function _onUpdateVms(vmobjs /* , server_uuid */) {
+                    var keys = Object.keys(vmobjs);
+
+                    t.ok(true, 'saw PUT /vms: (' + keys.length + ')');
+                    keys.forEach(function _putVmToVmapi(vm) {
+                        // ignore updates from VMs that existed when we started
+                        mocks.Vmapi.putVm(vmobjs[vm]);
+                    });
+                    cb();
+                }
+            );
+        }, function _createDoNotInventoryVm2(arg, cb) {
+            // creating a VM with DNI should not result in an update.
+            vmadm.create(payload2, function _vmadmCreateCb(err, info) {
+                t.ifError(err, 'create DNI VM2');
+                if (!err && info) {
+                    t.ok(info.uuid, 'new DNI VM2 has uuid: ' + info.uuid);
+                }
+                cb(err);
+            });
+        }, function _destroyVm1(arg, cb) {
+            //  With all modifications complete, we now delete the VM
+            vmadm.delete({
+                include_dni: true,
+                log: config.log,
+                uuid: payload1.uuid
+            }, function _vmadmDeleteCb(err) {
+                t.ifError(err, 'delete VM: '
+                    + (err ? err.message : 'success'));
+                cb(err);
+            });
+        }, function _destroyVm2(arg, cb) {
+            //  With all modifications complete, we now delete the VM
+            vmadm.delete({
+                include_dni: true,
+                log: config.log,
+                uuid: payload2.uuid
+            }, function _vmadmDeleteCb(err) {
+                t.ifError(err, 'delete VM: '
+                    + (err ? err.message : 'success'));
+                cb(err);
+            });
+        }, function _waitDestroyVm(arg, cb) {
+            setTimeout(function _waitedDestroyVm() {
+                t.ok(true, 'waited ' + WAIT_DNI_DELETE + ' ms after delete');
+                cb();
+            }, WAIT_DNI_DELETE);
+        }, function _checkUpdates(arg, cb) {
+            var badUpdates = 0;
+
+            updates.forEach(function _checkUpdate(vm) {
+                if (vm.uuid === payload1.uuid || vm.uuid === payload2.uuid) {
+                    badUpdates++;
+                }
+            });
+
+            t.equal(badUpdates, 0, 'no bad updates');
+            cb();
+        }
+    ]}, function _pipelineComplete(err) {
+        t.ifError(err, 'pipeline complete');
+        resetGlobalState(vmAgent); // so it's clean for the next test
+        t.end();
+    });
+
+    coordinator.on('vmapi.updateVm', function _onVmapiUpdateVm(vmobj, err) {
+        if (!err) {
+            mocks.Vmapi.putVm(vmobj);
+        }
+        updates.push(vmobj);
+    });
+
+    coordinator.on('vmadm.lookup', function _onVmadmLookup() {
+        t.fail('should not have seen vmadm.lookup, should have real vmadm');
```

Comment:
I'll add a comment:

```
        // vmadm.lookup is only emitted when the fakeVmadm is used, we're using
        // the real vmadm so seeing this would mean something's wrong with the
        // mocking.
```